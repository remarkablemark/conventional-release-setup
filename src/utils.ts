import child_process from 'child_process';

export const cwd = process.cwd();

const execSyncOptions = {
  cwd,
  stdio: 'inherit',
} as const;

/**
 * Runs command.
 */
export const exec = (
  command: string,
  options?: child_process.ExecSyncOptionsWithBufferEncoding,
) => child_process.execSync(command, { ...execSyncOptions, ...options });
