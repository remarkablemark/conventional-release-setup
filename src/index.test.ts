import fs from 'fs';
import path from 'path';
import { exec } from './utils';

const cwd = path.resolve(__dirname, '../lib');

function readFileSync(filepath: string) {
  return fs.readFileSync(path.resolve(cwd, filepath), 'utf8');
}

function clean() {
  fs.rmSync(cwd, { force: true, recursive: true });
}

describe('CLI', () => {
  beforeAll(() => {
    clean();
    exec('npm run build');
    exec('git init', { cwd });
    exec('npm init --yes', { cwd });
    exec('node index.js', { cwd });
  });

  afterAll(() => {
    clean();
  });

  it('updates package.json', () => {
    expect(JSON.parse(readFileSync('package.json'))).toMatchSnapshot();
  });

  it('copies .commitlintrc.json', () => {
    expect(JSON.parse(readFileSync('.commitlintrc.json'))).toMatchSnapshot();
  });

  it('copies .github/workflows/release-please.yml', () => {
    expect(
      readFileSync('.github/workflows/release-please.yml'),
    ).toMatchSnapshot();
  });

  it('adds .husky/.commit-msg', () => {
    expect(readFileSync('.husky/commit-msg')).toMatchSnapshot();
  });

  it('commits to git', () => {
    expect(
      exec('git --no-pager log -1', { cwd, stdio: 'pipe' }).toString(),
    ).toContain('chore: set up conventional release');
  });
});
