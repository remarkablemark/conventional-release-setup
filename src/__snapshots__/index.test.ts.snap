// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI adds .husky/.commit-msg 1`] = `
"npx commitlint --edit $1
"
`;

exports[`CLI copies .commitlintrc.json 1`] = `
{
  "extends": [
    "@commitlint/config-conventional",
  ],
}
`;

exports[`CLI updates package.json 1`] = `
{
  "author": "",
  "description": "",
  "devDependencies": {
    "@commitlint/cli": "^18.6.0",
    "@commitlint/config-conventional": "^18.6.0",
    "husky": "^9.0.6",
    "standard-version": "^9.5.0",
  },
  "keywords": [],
  "license": "ISC",
  "main": "index.js",
  "name": "lib",
  "scripts": {
    "prepare": "husky",
    "release": "standard-version --no-verify",
    "test": "echo "Error: no test specified" && exit 1",
  },
  "version": "1.0.0",
}
`;
